src/plugin/nerd_plugin.js

```diff
@@ -292,341 +355,139 @@ class NerdPluginVirtual extends EventEmitter {
     }
   }
 
-  _fulfillConditionLocal (transfer, fulfillment) {
-    if (!transfer) {
-      throw new Error('got transfer ID for nonexistant transfer')
-    } else if (!transfer.executionCondition) {
-      throw new Error('got transfer ID for OTP transfer')
-    }
-
-    return this.transferLog.isFulfilled(transfer.id).then((fulfilled) => {
-      if (fulfilled) {
-        throw new Error('this transfer has already been fulfilled')
-      } else {
-        return Promise.resolve(null)
-      }
-    }).then(() => {
-      const execute = transfer.executionCondition
-
-      const time = new Date()
-      const expiresAt = new Date(transfer.expiresAt)
-      const timedOut = (time > expiresAt)
-
-      if (this._validate(fulfillment, execute) && !timedOut) {
-        return this._executeTransfer(transfer, fulfillment)
-      } else if (timedOut) {
-        return this._timeOutTransfer(transfer)
-      } else {
-        throw new Error('invalid fulfillment')
-      }
-    }).catch(this._handle)
-  }
-
-  _executeTransfer (transfer, fulfillment) {
+  * _executeTransfer (transfer, fulfillment) {
     let fulfillmentBuffer = new Buffer(fulfillment)
     // because there is only one balance, kept, money is not _actually_ kept
     // in escrow (although it behaves as though it were). So there is nothing
     // to do for the execution condition.
-    return this.transferLog.fulfill(transfer.id, fulfillment).then(() => {
-      return this.transferLog.getDirection(transfer.id)
-    }).then((dir) => {
-      if (dir === this.transferLog.outgoing) {
-        this.emit('outgoing_fulfill', transfer, fulfillmentBuffer)
-        return this.balance.add(transfer.amount)
-      } else { // if (dir === this.transferLog.incoming)
-        this.emit('incoming_fulfill', transfer, fulfillmentBuffer)
-        return Promise.resolve(null)
-      }
-    }).then(() => {
-      return this.transferLog.getDirection(transfer.id)
-    }).then((dir) => {
-      return this.connection.send({
-        type: 'fulfill_execution_condition',
-        toNerd: (dir === this.transferLog.incoming),
-        transfer: transfer,
-        fulfillment: fulfillment
-      })
-    })
+
+    yield this.transferLog.fulfill(transfer.id, fulfillment)
+
+    const dir = yield this.transferLog.getDirection(transfer.id)
+    if (dir === this.transferLog.outgoing) {
+      this.emit('outgoing_fulfill', transfer, fulfillmentBuffer)
+      yield this.balance.add(transfer.amount)
+    } else { // if (dir === this.transferLog.incoming)
+      this.emit('incoming_fulfill', transfer, fulfillmentBuffer)
+    }
+
+    return {
+      transfer: transfer,
+      direction: (dir === this.transferLog.outgoing) ? 'incoming' : 'outgoing'
```

Comment:
changed to the full words instead. The ternary expression is still needed in order to switch the direction though.